### PR TITLE
Feature/credit card view

### PR DESCRIPTION
**Pull request by mateuszkula**
I want to merge this change because...
it adds credit card grid view.

Removed edit button from credit card tile as there wont be such option in saleor.

### CODE REVIEW

File: src/@next/components/molecules/CreditCardTile/stories.tsx [hunk 1]
```diff
@@ -11,12 +11,12 @@ const onRemove = action("onRemove called");
 const visa: CCProviders = "visa";
 
 const DEFAULT_PROPS = {
+  creditCardProvider: visa,
   expirationDate: "05/2019",
   last4Digits: 9876,
   nameOnCard: "John Doe",
   onEdit,
```
Comment (AlicjaSzu) [inline src/@next/components/molecules/CreditCardTile/stories.tsx]: it should be removed

File: src/@next/components/molecules/CreditCardTile/types.ts [hunk 2]
```diff
@@ -1,10 +1,12 @@
...
   expirationDate: string;
   last4Digits: number;
-  provider: CCProviders;
-  onRemove: () => void;
-  onEdit: () => void;
+  creditCardProvider: CCProviders;
+}
+export interface IProps extends ICreditCardData {
+  onRemove?: () => void;
+  onEdit?: () => void;
```
Comment (AlicjaSzu) [inline src/@next/components/molecules/CreditCardTile/types.ts]: not used anymore

File: src/@next/components/organisms/CreditCardGrid/CreditCardGrid.tsx [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+import { IProps } from "./types";
+
+export const CreditCardGrid: React.FC<IProps> = ({ creditCards }: IProps) => {
+  const addNewTile = [<AddNewTile type="card" />];
+  const ccTiles = creditCards.map(cc => {
+    return <CreditCardTile {...cc} />;
+  });
+
+  return <TileGrid elements={addNewTile.concat(ccTiles)} />;
```
Comment (AlicjaSzu) [inline src/@next/components/organisms/CreditCardGrid/CreditCardGrid.tsx]: columns prop is missing

Comment (AlicjaSzu) [reply src/@next/components/organisms/CreditCardGrid/CreditCardGrid.tsx]: or it should be optional

File: src/@next/components/organisms/CreditCardGrid/test.tsx [hunk 4]
```diff
@@ -0,0 +1,36 @@
+import { shallow } from "enzyme";
+import "jest-styled-components";
+import React from "react";
+
+import { CCProviders, TileGrid } from "@components/atoms";
+import { CreditCardGrid } from "./CreditCardGrid";
+import * as S from "./styles";
```
Comment (AlicjaSzu) [inline src/@next/components/organisms/CreditCardGrid/test.tsx]: not used

File: src/@next/components/organisms/CreditCardGrid/test.tsx [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+
+describe("<CreditCardGrid/>", () => {
+  const DEFAULT_PROPS = {};
```
Comment (AlicjaSzu) [inline src/@next/components/organisms/CreditCardGrid/test.tsx]: not used

File: src/@next/components/organisms/CreditCardGrid/test.tsx [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+
+  it("exists", () => {
+    const creditCardGrid = shallow(<CreditCardGrid creditCards={[]} />);
+
+    expect(creditCardGrid.exists()).toEqual(true);
+  });
+
+  it("should contain only AddNewTile if no credit cards provider", () => {
+    const creditCardGrid = shallow(<CreditCardGrid creditCards={[]} />);
+    expect(creditCardGrid.find(TileGrid).props().elements.length).toEqual(1);
```
Comment (AlicjaSzu) [inline src/@next/components/organisms/CreditCardGrid/test.tsx]: maybe we should check if element is equal to AddNewTile?

File: src/@next/components/organisms/CreditCardGrid/test.tsx [hunk 7]
```diff
@@ -0,0 +1,35 @@
+import { mount, shallow, ShallowWrapper } from "enzyme";
```
Comment (AlicjaSzu) [inline src/@next/components/organisms/CreditCardGrid/test.tsx]: ShallowWrapper is unused

File: src/@next/components/organisms/CreditCardGrid/CreditCardGrid.tsx [hunk 8]
```diff
@@ -0,0 +1,15 @@
+import React from "react";
+
+import { AddNewTile, TileGrid } from "@components/atoms";
+import { CreditCardTile } from "@components/molecules";
+
+import { IProps } from "./types";
+
+export const CreditCardGrid: React.FC<IProps> = ({ creditCards }: IProps) => {
+  const addNewTile = [<AddNewTile type="card" />];
+  const ccTiles = creditCards.map(cc => {
```
Comment (bogdal) [inline src/@next/components/organisms/CreditCardGrid/CreditCardGrid.tsx]: I would simplify the code and get rid of the `return` keyword from the next line.
```suggestion
  const ccTiles = creditCards.map(cc => (
```

**Pull request merged**